roxie/roxiemem/roxiemem.cpp

```diff
@@ -1695,7 +1777,7 @@ class FixedSizeHeaplet : public ChunkedHeaplet
             const char *block = data() + base;
             if (heapFlags & RHFnofragment)
             {
-                if (((std::atomic_uint *)block)->load(std::memory_order_relaxed) != FREE_ROW_COUNT)
+                if (((std::atomic_uint *)block)->load(std::memory_order_relaxed) < FREE_ROW_COUNT_MIN)
```

Comment:
This function is explicitly marked as potentially not giving correct results if there are concurrent allocations/releases, so I don't think code needs to change...  Ignoring that for the moment.

It would need to be acquire if this thread needed to ensure that all previous atomic items read from memory were consistent.  The only value it might depend on is freeBase.  You could get a false +ve if freeBase had been incremented before the link count had been set.  I wonder if that could potentially cause a problem elsewhere.

Only candidates are isFull() isEmpty() calcNumAllocated()

So possibly allocateSingle should set the value to zero, and then update freeBase with release semantics, and all accesses to freeBase should be acquire (if they are then scanning the memory bounded 0..freeBase).  Access to the linkcounts can remain relaxed.  What do you think @mckellyln?